metrics/docs/02_sign_and_notarize_metrics_binaries.md

```diff
@@ -0,0 +1,176 @@
+# Motivation
+> What problem is this project solving?
+
+Apple's ongoing initiatives at controlling what runs on their platforms took a new turn with macOS Catalina (10.15), with required app and command-line binary signing.
+
+[Codesign and notarize](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution) enables programs downloaded from the Internet to be opened without any warnings.
+
+To solve the described issue, the following steps is required:
+ - signing the binary with a Developer ID Certificate
+ - notarization (uploading the binary to Apple for approval)
+ - validating code signing
+ - validating notarization
+
+Currently, Metrics binaries for macOS have not signed or notarized.
+
+The document's goal is to investigate and analyze tools and methods that can help in signing and notarizing binaries.
+
+# References
+> Link to supporting documentation, GitHub tickets, etc.
+
+- [Notarizing macOS Software Before Distribution](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution)
+- [code signing for macOS](https://wiki.freepascal.org/Code_Signing_for_macOS)
+- [Using app-specific passwords](https://support.apple.com/en-us/HT204397)
+- [gon](https://github.com/mitchellh/gon)
+- [signing_tools](https://github.com/drud/signing_tools)
+- [notarize-cli](https://github.com/bacongravy/notarize-cli)
+- [Import codesign certificate Github Action](https://github.com/Apple-Actions/import-codesign-certs)
+
+# Analyze
+
+- [Requirements](#requirements)
+- [Analyzing process](#analyzing-process)
+- [Decision](#decision)
+
+## Requirements
+
+Before we can use tools, describes below, we need to:
+
+ - get an Apple ID
+ - generate an [app specific password](https://support.apple.com/en-us/HT204397)
+ - get the [Developer ID Application certificate](#developer-id-application)
+
+### Developer ID Application
+
+To get the certificate follow the next steps:
+ 1. On macOS open `Keychain Access` and select the `Keychain Access` button on the top left corner of the screen.
+ 2. Select the `Certificate Assistant` and then `Request a Certificate from a Certificate Authority`.
+ 3. Open the [following link](https://developer.apple.com/).
+ 4. Go to `Certificates IDs & Profiles`.
+ 5. Create a certificate.
+ 6. Choose the `Developer ID Application` from the list.
+ 7. Upload the signing request that we’ve created before.
+ 8. Download the created certificate and add it to your (default keychains) Keychain Access.
+ 9. Export the added certificate with the file format: `Personal information exchange(.p12)`.
+
+Extra step:
+
+To upload the exported certificate to the `Github Secrets`, we need to copy the base64 version:
+
+```
+base64 Certificates.p12 | pbcopy
+```
+
+And then paste the result to the `secrets`.
```

Comment:
I believe that this whole section should go to the `Design` part of the document - once we figure out an approach then we can outline the steps required to implement the approach. As you see that in the reality of it we stuck with the blocker https://github.com/dart-lang/sdk/issues/39106 - so that means that those steps not relevant at all - as we actually can't sign now and we need to go with sha256 approach for macs as well.